lib/src/file_watcher/polling.dart

```diff
@@ -50,13 +55,13 @@ class _PollingFileWatcher implements FileWatcher, ManuallyClosedWatcher {
 
   /// Checks the mtime of the file and whether it's been removed.
   Future<void> _poll() async {
-    // We don't mark the file as removed if this is the first poll (indicated by
-    // [_lastModified] being null). Instead, below we forward the dart:io error
-    // that comes from trying to read the mtime below.
+    // We don't mark the file as removed if this is the first poll. Instead,
+    // below we forward the dart:io error that comes from trying to read the
+    // mtime below.
     var pathExists = await File(path).exists();
     if (_eventsController.isClosed) return;
 
-    if (_lastModified != null && !pathExists) {
+    if (!_isFirstPoll && !pathExists) {
```

Comment:
I think this could instead just be `!isReady`? And then we can drop the extra field

Reply:
This change was actually a bug (I just pushed the fix) because here we meant to check "it did exist and now it doesn't". But in the other place it makes sense, will change :)